compiler/env/OMRAOTStorageInterface.hpp

```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+
+#ifndef OMR_AOTSTORAGEINTERFACE_INCL
+#define OMR_AOTSTORAGEINTERFACE_INCL
+
+#ifndef OMR_AOTSTORAGEINTERFACE_CONNECTOR
+#define OMR_AOTSTORAGEINTERFACE_CONNECTOR
+namespace OMR { class AOTStorageInterface; }
+namespace OMR { typedef OMR::AOTStorageInterface AOTStorageInterfaceConnector; }
+#endif // OMR_AOTSTORAGEINTERFACE_CONNECTOR
+
+#include "infra/Annotations.hpp"
+#include "env/AOTMethodHeader.hpp"
+#include "env/jittypes.h"
+
+namespace TR { class AOTStorageInterface; }
+namespace TR { class AOTMethodHeader; }
+
+namespace OMR {
+/**
+ * @brief The class is supposed to be inherited by 
+ * project level classes that implement two methods
+ * store Entry and load Entry, entry being a Method Header
+ * This will allow OMR AOT infrastructure to work with 
+ * preserved data 
+ */
+class OMR_EXTENSIBLE AOTStorageInterface
+   {
+public:
+   AOTStorageInterface(){};
+   /**
+    * @brief required by the extensible class
+    * 
+    * @return TR::AOTStorageInterface* 
+    */
+   TR::AOTStorageInterface* self();
```

Comment:
The `self()` function should be `protected`, not `public`.

Reply:
Hi, @mstoodle , I will fix that across the classes I am contributing.  For writing my code, I was looking at the rest of the compiler code, and I believe code snippets like https://github.com/eclipse/omr/blob/7b4155a69859e4422f742acbffe712d1d15ee4cc/compiler/compile/OMRCompilation.hpp#L288-L312 made me confused.

Reply:
that's another case where `self()` should almost certainly be made `protected`

Reply:
@mstoodle  Does the  ```inline`` keyword change the access modifier? I think there are a lot of classes that have self() marked  as public, OMRCodeGenerator, generic version has this problem too.

Reply:
created #5780 to look into that and other examples. Please make `self()` in your code `protected`.

Reply:
Will absolutely do that! Thank you very much!